cedar-policy-validator/src/schema.rs

```diff
@@ -2571,23 +2767,1487 @@ mod test {
             }
         });
         let schema = ValidatorSchema::from_json_value(src.clone(), Extensions::all_available());
-        assert_matches!(schema, Err(SchemaError::CommonTypeNameConflict(CommonTypeNameConflictError(n))) if n == "String".parse().unwrap());
+        assert_matches!(schema, Ok(_));
     }
 }
 
+/// Tests described in https://github.com/cedar-policy/cedar/issues/579
+///
+/// We test all possible (position, scenario) pairs where:
+///
+/// position is all places a typename can occur in a schema:
+/// A. Inside a context attribute type
+/// B. Inside an entity attribute type
+/// C. Inside the body of a common-type definition
+/// D. As an entity parent type
+/// E. In an action `appliesTo` declaration
+/// F. Inside an action attribute type?
+/// G. In an action parent declaration?
```

Comment:
There are only tests for A-C right now actually, as noted in the PR description.  I believe the current code will fail some of the D-G tests.

Reply:
Added tests for D and E.  (D tests required code changes -- the first version of this PR didn't apply the same resolution algorithm to entity parent declarations.)  F tests don't make sense because action attributes are not _typenames_, they are actual _attribute values_, so we can't use the same resolution algorithm.  And also, action attributes are not supported in the human syntax, only the JSON syntax.  I renamed G to F and am working on that.  But that one is subtle too -- G requires not that the _typenames_ conditionally resolve, but that _action EUIDs_ conditionally resolve.  Not based on whether the typename `NS1::Action` exists, but whether `NS1::Action::"foo"` was declared.  It's unclear whether #579 envisioned this.  Do we want `action Read in [Group]` to always unconditionally refer to `NS::Action::"Group"` where `NS` is the current namespace (current behavior)?  Or do we want it to conditionally resolve to `Action::"Group"` if `NS::Action::"Group"` doesn't exist?

Reply:
I'm going to proceed assuming the latter

Reply:
I think that makes sense.

Reply:
Pushed the F tests, and they pass.  This required some medium-weight code changes to support resolving action EUIDs, not just typenames.